deploy/terraform/modules/app_tier/vm-app.tf

```
@@ -14,9 +14,9 @@ resource "azurerm_network_interface" "app" {
   }
 }
 
-# Create the Application VM(s)
+# Create the Linux Application VM(s)
 resource "azurerm_linux_virtual_machine" "app" {
-  count                        = local.enable_deployment ? local.application_server_count : 0
+  count                        = local.enable_deployment ? (local.app_ostype == "Linux" ? local.application_server_count : 0) : 0
```

For all `ostype` condition, please use upper to convert the variable to upper case (there are a couple of other places below).
This is to handle variation in the input like "Linux/linux/LINUX" or "Windows/windows/WINDOWS".
```suggestion
  count                        = local.enable_deployment ? (upper(local.app_ostype) == "LINUX" ? local.application_server_count : 0) : 0
```